### ISSUE TITLE
Allow identical public SSH key for multiple repos

### ISSUE THREAD

**Issue by Forceu** [Ravinou/borgwarehouse#236]
In my use case, I have a central server that runs multiple borg backups to an external server. As the server is managing multiple repos, I would require it to generate a new SSH key for every repo, which is not ideal.

It would be great to be able to re-use the public keys. If you feel that it is a security threat, you could maybe add an environment variable that the user has to set in order to disable this check?

**Comment by Ravinou** [comment]
You will find the answer in the documentation, [here](https://borgwarehouse.com/docs/admin-manual/how-to-update/#whats-new-in-v2-).

**Comment by Forceu** [comment]
@Ravinou I just had the time to write an application to work as a custom SSH server. This would have the following advantages:

- Will work as a non-root service (also closes #370 and #401)
- Can map multiple public keys to a single repo and one key to multiple repos (would fix this ticket, #87, #147 and #154)

It would also give the option to assign different permissions to keys. E.g. key A could have full write permissions on repo X, whereas key B can only use append-only mode

The application is written in Go and therefore platform-independent. On Linux it is about 9MB in size.

I have uploaded the proof-of-concept code here: https://github.com/Forceu/borgwarehouseSSH/blob/main/main.go

Basically it is a middleware for an already existing and proven SSH server. Instead of opening an interactive session, it only allows `borg serve`, the same as the current instance of OpenSSH.

The middleware parses the `/home/borgwarehouse/.ssh/authorized_keys` file and allows login in the format [`reponame]@[server]` with the given public keys.

Like I said, this is only a proof of concept and the coding style is not that great at the moment. If you don't have any experience with Go, I can provide you with the binary or a docker image. Would you be interested in something like this or rather stick with OpenSSH/Dropbear?

**Comment by Ravinou** [comment]
Don't you think we could manage to provide several docker images: 

- openssh
- dropbear
- yourapplication

I think that's where we need to go to keep everyone happy. What do you think?
